lib/logstash/filters/grok.rb

```diff
@@ -404,16 +408,34 @@ def add_patterns_from_files(paths, grok)
       end
     end # def add_patterns_from_files
 
-    private
     def add_patterns_from_inline_definition(pattern_definitions, grok)
       pattern_definitions.each do |name, pattern|
         next if pattern.nil?
         grok.add_pattern(name, pattern.chomp)
       end
     end
 
-    def close
-      @timeout_enforcer.stop!
-    end
+    class TimeoutError < RuntimeError; end
+
+    class GrokException < Exception
+      attr_reader :grok, :field, :value
 
+      def initialize(grok = nil, field = nil, value = nil)
+        @field = field
+        @value = value
+        @grok = grok
```

Comment:
Nitpick: when using positional arguments to instantiate an object with named properties, assigning those properties in the order they are provided reads a little nicer.